### PR TITLE
Use `ubuntu-latest` in PyPI publish CI

### PR DESCRIPTION
**Pull request by not522**
## Motivation
The PyPI publish CI has used ubuntu-18.04, which is deprecated.

## Description of the changes
- Use `ubuntu-latest` in PyPI publish CI

### CODE REVIEW

**Comment by toshihikoyanase** [comment]
@cross32768 Could you review this PR, please?

@not522 The pypi publish jobs are not triggered by PR. Do you think we should check the behavior before wer merge the change? Or, we can fix it if the daily CI failed after merging this PR?

**Comment by not522** [comment]
> Do you think we should check the behavior before wer merge the change?

It's reasonable. I'll check that we can upload it to TestPyPI.

**Review by toshihikoyanase — APPROVED** [review]
I confirmed that optuna packages could be built on `ubuntu:22.04` using docker. They passed `twine check`.

And I also found that `twine` is tested on `ubuntu-latest`.
https://github.com/pypa/twine/blob/9c2c0a1c535155931c3d879359330cb836950c6a/.github/workflows/main.yml#L46

So, I think testing this PR on testPyPI may be over-engineering, and I'd like to approve this PR.

**Review by cross32768 — APPROVED** [review]
I also confirmed that `twine check` returns "passed" by the process shown below (maybe the same process as above reviewer). LGTM!

* `docker run -it ubuntu:22.04`
* install required dependencies (`apt install git`, `pip install twine`, etc)
* `git clone https://github.com/optuna/optuna`
* `python3 -m build`
* `twine check dist/optuna-3.2.0.dev0-py3-none-any.whl`

It returns:
`Checking dist/optuna-3.2.0.dev0-py3-none-any.whl: PASSED
`

**Comment by toshihikoyanase** [comment]
Let me merge this PR since it got two approvals. Thank you.

**Pull request merged**
